### ISSUE TITLE
[4.1]: List of orders fails to load

### ISSUE THREAD

**Issue by ul8** [craftcms/commerce#2936]
### Description

I'm updating an old Craft installation (3.3 with Commerce 2) to Craft 4 and Commerce 4.

Craft 3.7.x + Commerce 3 worked fine, but further upgrades to Craft 4 and Commerce 4 breaks the list of orders. 

### Steps to reproduce

No other steps other than updating the installation, logging in and accessing the orders page.

During the update, I followed the instructions here: https://craftcms.com/docs/commerce/4.x/upgrading.html#performing-the-upgrade

### Expected behavior

The orders page should load

### Actual behavior

/admin/commerce/orders fails to load (stays blank). I'm seeing this in network dev tools:

Request: 

https://REDACTED/index.php?p=admin%2Factions%2Felement-indexes%2Fget-elements&v=1660568780394

Response: 
```
{"message":"craft\\commerce\\models\\LineItem::setDescription(): Argument #1 ($description) must be of type string, null given, called in /app/vendor/yiisoft/yii2/base/Component.php on line 180","exception":"TypeError","file":"/app/vendor/craftcms/commerce/src/models/LineItem.php","line":322,"trace":[{"file":"/app/vendor/yiisoft/yii2/base/Component.php","line":180,"function":"setDescription","class":"craft\\commerce\\models\\LineItem","type":"->"},{"file":"/app/vendor/yiisoft/yii2/BaseYii.php","line":558,"function":"__set","class":"yii\\base\\Component","type":"->"},{"file":"/app/vendor/yiisoft/yii2/base/BaseObject.php","line":107,"function":"configure","class":"yii\\BaseYii","type":"::"},{"file":"/app/vendor/craftcms/cms/src/base/Model.php","line":78,"function":"__construct","class":"yii\\base\\BaseObject","type":"->"},{"file":"/app/vendor/craftcms/commerce/src/services/LineItems.php","line":158,"function":"__construct","class":"craft\\base\\Model","type":"->"},{"file":"/app/vendor/craftcms/commerce/src/elements/Order.php","line":2578,"function":"getAllLineItemsByOrderId","class":"craft\\commerce\\services\\LineItems","type":"->"},{"file":"/app/vendor/craftcms/commerce/src/elements/Order.php","line":2700,"function":"getLineItems","class":"craft\\commerce\\elements\\Order","type":"->"},{"file":"/app/vendor/craftcms/commerce/src/elements/Order.php","line":2407,"function":"getItemSubtotal","class":"craft\\commerce\\elements\\Order","type":"->"},{"file":"/app/vendor/craftcms/commerce/src/elements/Order.php","line":2454,"function":"getTotalPrice","class":"craft\\commerce\\elements\\Order","type":"->"},{"file":"/app/vendor/craftcms/commerce/src/elements/Order.php","line":2461,"function":"getOutstandingBalance","class":"craft\\commerce\\elements\\Order","type":"->"},{"file":"/app/vendor/craftcms/commerce/src/elements/Order.php","line":2277,"function":"hasOutstandingBalance","class":"craft\\commerce\\elements\\Order","type":"->"},{"file":"/app/vendor/craftcms/commerce/src/elements/Order.php","line":2336,"function":"getIsPaid","class":"craft\\commerce\\elements\\Order","type":"->"},{"file":"/app/vendor/craftcms/commerce/src/elements/Order.php","line":2382,"function":"getPaidStatus","class":"craft\\commerce\\elements\\Order","type":"->"},{"file":"/app/vendor/craftcms/commerce/src/elements/traits/OrderElementTrait.php","line":120,"function":"getPaidStatusHtml","class":"craft\\commerce\\elements\\Order","type":"->"},{"file":"/app/vendor/craftcms/cms/src/base/Element.php","line":4181,"function":"tableAttributeHtml","class":"craft\\commerce\\elements\\Order","type":"->"},{"file":"/app/vendor/twig/twig/src/Extension/CoreExtension.php","line":1607,"function":"getTableAttributeHtml","class":"craft\\base\\Element","type":"->"},{"file":"/app/vendor/craftcms/cms/src/helpers/Template.php","line":101,"function":"twig_get_attribute"},{"file":"/app/storage/runtime/compiled_templates/39/39b760b27abe3f400c0dab0ea0ef6dc6.php","line":182,"function":"attribute","class":"craft\\helpers\\Template","type":"::"},{"file":"/app/vendor/twig/twig/src/Template.php","line":394,"function":"doDisplay","class":"__TwigTemplate_f3e0e1f92072513bb2f61648505ab858","type":"->"},{"file":"/app/vendor/twig/twig/src/Template.php","line":367,"function":"displayWithErrorHandling","class":"Twig\\Template","type":"->"},{"file":"/app/storage/runtime/compiled_templates/dc/dc4924fe5199e9a31070ca0bf67fe519.php","line":127,"function":"display","class":"Twig\\Template","type":"->"},{"file":"/app/vendor/twig/twig/src/Template.php","line":394,"function":"doDisplay","class":"__TwigTemplate_181b2ce35d57e680308162a5f133fd2d","type":"->"},{"file":"/app/vendor/twig/twig/src/Template.php","line":367,"function":"displayWithErrorHandling","class":"Twig\\Template","type":"->"},{"file":"/app/vendor/twig/twig/src/Template.php","line":379,"function":"display","class":"Twig\\Template","type":"->"},{"file":"/app/vendor/twig/twig/src/TemplateWrapper.php","line":40,"function":"render","class":"Twig\\Template","type":"->"},{"file":"/app/vendor/twig/twig/src/Environment.php","line":277,"function":"render","class":"Twig\\TemplateWrapper","type":"->"},{"file":"/app/vendor/craftcms/cms/src/web/View.php","line":428,"function":"render","class":"Twig\\Environment","type":"->"},{"file":"/app/vendor/craftcms/cms/src/base/Element.php","line":1064,"function":"renderTemplate","class":"craft\\web\\View","type":"->"},{"file":"/app/vendor/craftcms/cms/src/controllers/ElementIndexesController.php","line":649,"function":"indexHtml","class":"craft\\base\\Element","type":"::"},{"file":"/app/vendor/craftcms/cms/src/controllers/ElementIndexesController.php","line":146,"function":"elementResponseData","class":"craft\\controllers\\ElementIndexesController","type":"->"},{"function":"actionGetElements","class":"craft\\controllers\\ElementIndexesController","type":"->"},{"file":"/app/vendor/yiisoft/yii2/base/InlineAction.php","line":57,"function":"call_user_func_array"},{"file":"/app/vendor/yiisoft/yii2/base/Controller.php","line":178,"function":"runWithParams","class":"yii\\base\\InlineAction","type":"->"},{"file":"/app/vendor/yiisoft/yii2/base/Module.php","line":552,"function":"runAction","class":"yii\\base\\Controller","type":"->"},{"file":"/app/vendor/craftcms/cms/src/web/Application.php","line":301,"function":"runAction","class":"yii\\base\\Module","type":"->"},{"file":"/app/vendor/craftcms/cms/src/web/Application.php","line":625,"function":"runAction","class":"craft\\web\\Application","type":"->"},{"file":"/app/vendor/craftcms/cms/src/web/Application.php","line":280,"function":"_processActionRequest","class":"craft\\web\\Application","type":"->"},{"file":"/app/vendor/yiisoft/yii2/base/Application.php","line":384,"function":"handleRequest","class":"craft\\web\\Application","type":"->"},{"file":"/app/web/index.php","line":23,"function":"run","class":"yii\\base\\Application","type":"->"}],"error":"craft\\commerce\\models\\LineItem::setDescription(): Argument #1 ($description) must be of type string, null given, called in /app/vendor/yiisoft/yii2/base/Component.php on line 180"}
```

### Craft CMS version

Craft Pro 4.2.1.1

### Craft Commerce version

4.1.0

### PHP version

8.1.8

### Operating system and version

Linux 5.4.0-113-generic

### Database type and version

PostgreSQL 13.6

### Installed plugins and versions

CP Field Inspect    1.4.4
Craft Commerce  4.1.0
Enupal Snapshot 2.0.1
Environment Label   4.0.2
Imager X    4.1.2
Incognito Field 1.3.0
Redactor    3.0.2
Stripe for Craft Commerce   3.0.1
Super Table 3.0.0
Typed link field    2.1.3-rc
User Group Field    3.0.1

**Comment by lukeholder** [comment]
You can fix this now by running this sql command (after updating):

```sql
UPDATE commerce_lineitems
SET description = ''
WHERE description is null;
```

We will leave this ticket open while we make a fix to the upgrade migration. Thanks.

**Comment by ul8** [comment]
Thanks for the quick reply Luke! That fixed the error, but I was getting another one:
```
{"message":"craft\\commerce\\models\\LineItem::setSku(): Argument #1 ($sku) must be of type string, null given, called in /app/vendor/yiisoft/yii2/base/Component.php on line 180","exception":"TypeError","file":"/app/vendor/craftcms/commerce/src/models/LineItem.php","line":337,"trace":[{"file":"/app/vendor/yiisoft/yii2/base/Component.php","line":180,"function":"setSku","class":"craft\\commerce\\models\\LineItem","type":"->"},{"file":"/app/vendor/yiisoft/yii2/BaseYii.php","line":558,"function":"__set","class":"yii\\base\\Component","type":"->"},{"file":"/app/vendor/yiisoft/yii2/base/BaseObject.php","line":107,"function":"configure","class":"yii\\BaseYii","type":"::"},{"file":"/app/vendor/craftcms/cms/src/base/Model.php","line":78,"function":"__construct","class":"yii\\base\\BaseObject","type":"->"},{"file":"/app/vendor/craftcms/commerce/src/services/LineItems.php","line":158,"function":"__construct","class":"craft\\base\\Model","type":"->"},{"file":"/app/vendor/craftcms/commerce/src/elements/Order.php","line":2578,"function":"getAllLineItemsByOrderId","class":"craft\\commerce\\services\\LineItems","type":"->"},{"file":"/app/vendor/craftcms/commerce/src/elements/Order.php","line":2700,"function":"getLineItems","class":"craft\\commerce\\elements\\Order","type":"->"},{"file":"/app/vendor/craftcms/commerce/src/elements/Order.php","line":2407,"function":"getItemSubtotal","class":"craft\\commerce\\elements\\Order","type":"->"},{"file":"/app/vendor/craftcms/commerce/src/elements/Order.php","line":2454,"function":"getTotalPrice","class":"craft\\commerce\\elements\\Order","type":"->"},{"file":"/app/vendor/craftcms/commerce/src/elements/Order.php","line":2461,"function":"getOutstandingBalance","class":"craft\\commerce\\elements\\Order","type":"->"},{"file":"/app/vendor/craftcms/commerce/src/elements/Order.php","line":2277,"function":"hasOutstandingBalance","class":"craft\\commerce\\elements\\Order","type":"->"},{"file":"/app/vendor/craftcms/commerce/src/elements/Order.php","line":2336,"function":"getIsPaid","class":"craft\\commerce\\elements\\Order","type":"->"},{"file":"/app/vendor/craftcms/commerce/src/elements/Order.php","line":2382,"function":"getPaidStatus","class":"craft\\commerce\\elements\\Order","type":"->"},{"file":"/app/vendor/craftcms/commerce/src/elements/traits/OrderElementTrait.php","line":120,"function":"getPaidStatusHtml","class":"craft\\commerce\\elements\\Order","type":"->"},{"file":"/app/vendor/craftcms/cms/src/base/Element.php","line":4181,"function":"tableAttributeHtml","class":"craft\\commerce\\elements\\Order","type":"->"},{"file":"/app/vendor/twig/twig/src/Extension/CoreExtension.php","line":1607,"function":"getTableAttributeHtml","class":"craft\\base\\Element","type":"->"},{"file":"/app/vendor/craftcms/cms/src/helpers/Template.php","line":101,"function":"twig_get_attribute"},{"file":"/app/storage/runtime/compiled_templates/39/39b760b27abe3f400c0dab0ea0ef6dc6.php","line":182,"function":"attribute","class":"craft\\helpers\\Template","type":"::"},{"file":"/app/vendor/twig/twig/src/Template.php","line":394,"function":"doDisplay","class":"__TwigTemplate_f3e0e1f92072513bb2f61648505ab858","type":"->"},{"file":"/app/vendor/twig/twig/src/Template.php","line":367,"function":"displayWithErrorHandling","class":"Twig\\Template","type":"->"},{"file":"/app/storage/runtime/compiled_templates/dc/dc4924fe5199e9a31070ca0bf67fe519.php","line":127,"function":"display","class":"Twig\\Template","type":"->"},{"file":"/app/vendor/twig/twig/src/Template.php","line":394,"function":"doDisplay","class":"__TwigTemplate_181b2ce35d57e680308162a5f133fd2d","type":"->"},{"file":"/app/vendor/twig/twig/src/Template.php","line":367,"function":"displayWithErrorHandling","class":"Twig\\Template","type":"->"},{"file":"/app/vendor/twig/twig/src/Template.php","line":379,"function":"display","class":"Twig\\Template","type":"->"},{"file":"/app/vendor/twig/twig/src/TemplateWrapper.php","line":40,"function":"render","class":"Twig\\Template","type":"->"},{"file":"/app/vendor/twig/twig/src/Environment.php","line":277,"function":"render","class":"Twig\\TemplateWrapper","type":"->"},{"file":"/app/vendor/craftcms/cms/src/web/View.php","line":428,"function":"render","class":"Twig\\Environment","type":"->"},{"file":"/app/vendor/craftcms/cms/src/base/Element.php","line":1064,"function":"renderTemplate","class":"craft\\web\\View","type":"->"},{"file":"/app/vendor/craftcms/cms/src/controllers/ElementIndexesController.php","line":649,"function":"indexHtml","class":"craft\\base\\Element","type":"::"},{"file":"/app/vendor/craftcms/cms/src/controllers/ElementIndexesController.php","line":146,"function":"elementResponseData","class":"craft\\controllers\\ElementIndexesController","type":"->"},{"function":"actionGetElements","class":"craft\\controllers\\ElementIndexesController","type":"->"},{"file":"/app/vendor/yiisoft/yii2/base/InlineAction.php","line":57,"function":"call_user_func_array"},{"file":"/app/vendor/yiisoft/yii2/base/Controller.php","line":178,"function":"runWithParams","class":"yii\\base\\InlineAction","type":"->"},{"file":"/app/vendor/yiisoft/yii2/base/Module.php","line":552,"function":"runAction","class":"yii\\base\\Controller","type":"->"},{"file":"/app/vendor/craftcms/cms/src/web/Application.php","line":301,"function":"runAction","class":"yii\\base\\Module","type":"->"},{"file":"/app/vendor/craftcms/cms/src/web/Application.php","line":625,"function":"runAction","class":"craft\\web\\Application","type":"->"},{"file":"/app/vendor/craftcms/cms/src/web/Application.php","line":280,"function":"_processActionRequest","class":"craft\\web\\Application","type":"->"},{"file":"/app/vendor/yiisoft/yii2/base/Application.php","line":384,"function":"handleRequest","class":"craft\\web\\Application","type":"->"},{"file":"/app/web/index.php","line":23,"function":"run","class":"yii\\base\\Application","type":"->"}],"error":"craft\\commerce\\models\\LineItem::setSku(): Argument #1 ($sku) must be of type string, null given, called in /app/vendor/yiisoft/yii2/base/Component.php on line 180"}
```
fixed it with 

```
UPDATE commerce_lineitems
SET sku = ''
WHERE sku IS NULL;
```

**Comment by billythekid** [comment]
I ran these commands and both updated 7 lines, probably the same ones.

All the previously erroring pages now load. Looks like I had some rogue old orders on there in a weird state because whatever that did seems to have disappeared them from the DB. This comes up empty ¯\\\_(ツ)_/¯

```SQL
select * from commerce_lineitems
where description = '' or sku = '';
```

**Comment by jamiepittock** [comment]
Has there been a regression? We've come across this issue during some Commerce 4 upgrade testing. Description and SKU. Fixes above worked.

**Comment by lukeholder** [comment]
@jamiepittock I’ll look into it again now.

**Comment by lukeholder** [comment]
Just added a fix, which will be in the next release. I will let you know once it’s out.

To get the fix early, change your `craftcms/commerce` requirement in composer.json to:

```json
"require": {
  "craftcms/cms": "dev-develop#62953a4dce09e44f055e836f0a58863d5f24acd6",
  "...": "..."
}
```

Then run `composer update`.

**Comment by lukeholder** [comment]
This should be fixed in 4.2.5.1
